src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Extensions/StartupExtensions.cs

```diff
@@ -46,34 +49,56 @@ public static IServiceCollection RegisterDatabase(this IServiceCollection servic
 
         public static IServiceCollection RegisterServices(this IServiceCollection services, IConfiguration configuration)
         {
-            services.AddSingleton<IEmailSender, EmailSender>(
-                sender =>
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            services.AddSingleton<IEmailSender, EmailSender>(sender =>
+            {
+                var emailSender = new EmailSender
                 {
-                    var emailSender = new EmailSender
-                    {
-                        Host = configuration.GetValue<string>("Email:Smtp:Host"),
-                        Port = configuration.GetValue<int>("Email:Smtp:Port"),
-                        UserName = configuration.GetValue<string>("Email:Smtp:UserName"),
-                        Password = configuration.GetValue<string>("Email:Smtp:Password"),
-                        From = configuration.GetValue<string>("Email:Smtp:From"),
-                        Ssl = configuration.GetValue<bool?>("Email:Smtp:Ssl") ?? false,
-                    };
-
-                    return emailSender;
-                });
+                    Host = configuration.GetValue<string>("Email:Smtp:Host"),
+                    Port = configuration.GetValue<int>("Email:Smtp:Port"),
+                    UserName = configuration.GetValue<string>("Email:Smtp:UserName"),
+                    Password = configuration.GetValue<string>("Email:Smtp:Password"),
+                    From = configuration.GetValue<string>("Email:Smtp:From"),
+                    Ssl = configuration.GetValue<bool?>("Email:Smtp:Ssl") ?? false,
+                };
+
+                return emailSender;
+            });
+
+            services.AddAutoMapper(typeof(AdminService));
```

Comment:
why AdminService? the code was
`.AddAutoMapper(typeof(Startup))`

Reply:
Because, we have mapper profiles in Services project